Mathlib/Order/Basic.lean

```diff
@@ -1172,6 +1172,27 @@ lemma mk_lt_mk_of_le_of_lt (h₁ : a₁ ≤ a₂) (h₂ : b₁ < b₂) : (a₁,
 
 end Preorder
 
+section PartialOrder
+
+variable [PartialOrder α] [Preorder β] {x y : α × β}
+
+lemma toLex (h : x < y) : Prod.Lex (· < ·) (· < ·) x y := by
+  cases y
+  cases' lt_iff.1 h with h₁ h₂
+  · exact Lex.left _ _ h₁.1
+  · obtain h₂ | rfl := lt_or_eq_of_le h₂.1
+    · exact Lex.left _ _ h₂
+    · exact Lex.right _ h₂.2
```

Comment:
Is this related to

https://github.com/leanprover-community/mathlib4/blob/48fd37e25593c8ea6b65bd3d60a4330cdd12d40d/Mathlib/Data/Prod/Lex.lean#L112-L116

somehow?

Reply:
I think this was the result I was looking for earlier. I'll edit the PR later to only include the `lt_of_le_of_lt` result.